pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authorization/PulsarAuthorizationProvider.java

```diff
@@ -536,17 +536,32 @@ private void validatePoliciesReadOnlyAccess() {
             case CLEAR_BACKLOG:
                 isAuthorizedFuture = allowConsumeOpsAsync(namespaceName, role, authData);
                 break;
+            case CREATE_TOPIC:
+            case DELETE_TOPIC:
+            case ADD_BUNDLE:
+            case GET_BUNDLE:
+            case DELETE_BUNDLE:
+            case GRANT_PERMISSION:
+            case GET_PERMISSION:
+            case REVOKE_PERMISSION:
+                return validateTenantAdminAccess(namespaceName.getTenant(), role, authData);
             default:
-                isAuthorizedFuture = CompletableFuture.completedFuture(false);
+                return FutureUtil.failedFuture(
+                        new IllegalStateException("NamespaceOperation [" + operation.name() + "] is not supported."));
         }
-        CompletableFuture<Boolean> isTenantAdminFuture = validateTenantAdminAccess(namespaceName.getTenant(), role, authData);
-        return isTenantAdminFuture.thenCombine(isAuthorizedFuture, (isTenantAdmin, isAuthorized) -> {
-            if (log.isDebugEnabled()) {
-                log.debug("Verify if role {} is allowed to {} to topic {}: isTenantAdmin={}, isAuthorized={}",
-                        role, operation, namespaceName, isTenantAdmin, isAuthorized);
-            }
-            return isTenantAdmin || isAuthorized;
-        });
+
+        return validateTenantAdminAccess(namespaceName.getTenant(), role, authData)
+                .thenCompose(isSuperUserOrAdmin -> {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Verify if role {} is allowed to {} to namespace {}: isSuperUserOrAdmin={}",
+                                role, operation, namespaceName, isSuperUserOrAdmin);
+                    }
+                    if (isSuperUserOrAdmin) {
+                        return CompletableFuture.completedFuture(true);
+                    } else {
+                        return isAuthorizedFuture;
```

Comment:
why are we still creating `isAuthorizedFuture` ?

if we are `SuperUserOrAdmin` we can skip evaluating the logic for `isAuthorizedFuture`